crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -4379,16 +4376,13 @@ impl<'a> Checker<'a> {
             .scopes
             .ancestors(scope_id)
             .skip(1)
+            .filter(|scope| scope.kind.is_function() || scope.kind.is_module())
             .find_map(|scope| scope.get(name))
         {
             // Otherwise, if there's an existing binding in a parent scope, mark it as shadowed.
-            let binding = self.semantic.binding(binding_id);
-            let shadowed = self.semantic.binding(shadowed_id);
-            if binding.redefines(shadowed) {
```

Comment:
This isn't necessary -- we do this later, in the `RedefinedWhileUnused` rule, and it's actually specific to that rule, not to shadowing.